docs/en/serial-vault.md

```diff
@@ -0,0 +1,415 @@
+securely to the Serial Vault.
+
+### Generating a key
+
+Generate the keys for signing serial and system-user assertions as follows:
+
+!!! Note:
+          Keys used for brand purposes are critically important assets that
+          should not be lost. Snapcraft keys are created in ~/.snap/gnupg/.
+          Ensure you back this directory up to a secure external medium.
+          Registering the key does not upload the key and therefore does not
+          support key recovery.
+
+```bash
+snap create-key serial 
+```
+
+You will be prompted for a passphrase. In this case, the passphrase **MUST BE BLANK** so just press `Return` twice.
+
+Note that it takes some time to create the key. 
+
+After completion, verify creation with the following command:
+
+```bash
+snapcraft list-keys
+```
+This will return a list of the known keys. At the moment, the key you just added will be marked as 'not registered'.
+
+```bash
+    Name 	SHA3-384 fingerprint
+-   serial   yOkSks8x1OICgx60WW-YEKehTeWvQpE2qA17ZM-RT8i0u7YYrIOjSp1DARkv4Cuu 
+(not registered)
+```
+
+### Registering the key
+
+Register the key to the Brand SSO account as follows:
+
+```bash
+snapcraft register-key serial
+```
+
+You can confirm the key has been added with the command:
+
+```bash
+snapcraft list-keys
+```
+...which will now list the key with an asterisk (`*`) denoting that it is registered:
+
+```bash
+    Name 	SHA3-384 fingerprint
+*   serial   yOkSks8x1OICgx60WW-YEKehTeWvQpE2qA17ZM-RT8i0u7YYrIOjSp1DARkv4Cuu
+```
+
+### Exporting the key
+
+Before uploading the key, it must be encrypted. Export the key using ASCII
+armored encryption, as follows:
+ 
+```bash
+gpg --homedir ~/.snap/gnupg --armor --export-secret-key serial > serial.asc
+```
+
+In this example, serial.asc is the serial key file to upload to Serial Vault.
+You name a key used to sign system-user assertions differently.
+
+### Uploading the key to the Serial Vault
+
+1. Login to the Serial Vault as the Brand SSO account.
+1. Select Models.
+
+
+
+![image][serial-vault-image2]
+1. Click the Signing Keys + button.
+
+
+
+![image][serial-vault-image3]
+
+
+
+![image][serial-vault-image4]
+
+1. Enter account-id of the Brand SSO account as the Signing Authority.
+1. Under Signing Key, click Browse, then find and select the previously
+created ASCII-armored serial key file.. 
+1. Click Save. 
+The list of models and signing keys displays, including this new one.
+
+## Creating a model
+
+You need to create a model. This corresponds to the model previously used when
+building an image. (Every system has a model.) At runtime, the system sends
+the model to the Serial Vault, which uses it to authenticate the system before
+generating a serial assertion.
+
+The model contains key data that must exactly match data in the model in the
+deployed system, including:
+
+ - brand-id: this is the account-id of the Brand SSO account
+ - model name
+
+Also, the system itself must have a gadget snap whose prepare-device hook
```

Comment:
Link to https://docs.ubuntu.com/core/en/guides/build-device/config-hooks#supported-hooks for "prepare-device hook".